client/packages/coldchain/src/Equipment/ListView/AddFromScannerButton.tsx

```diff
@@ -30,19 +34,60 @@ export const AddFromScannerButtonComponent = () => {
   const equipmentRoute = RouteBuilder.create(AppRoute.Coldchain).addPart(
     AppRoute.Equipment
   );
-  const { mutateAsync: fetchAsset } = useAssets.document.fetch();
+  const { mutateAsync: scanAsset } = useAssets.document.scan();
+  const { mutateAsync: saveNewAsset } = useAssets.document.insert();
+  const { insertLog, invalidateQueries } = useAssets.log.insert();
+  const newAssetData = useRef<DraftAsset>();
+
+  const showCreateConfirmation = useConfirmationModal({
+    onConfirm: () => {
+      if (newAssetData.current) {
+        saveNewAsset(newAssetData.current)
+          .catch(e => error(t('error.unable-to-save-asset', { error: e }))())
+          .then(async () => {
+            if (newAssetData.current) {
+              await insertLog({
```

Comment:
This is duplicated functionality as the add new asset modal does this too.
Would be nice to consolodate in #5392